worker/peergrouper/initiate.go

```diff
@@ -50,17 +51,23 @@ func InitiateMongoServer(p InitiateMongoParams) error {
 
 	// Initiate may fail while mongo is initialising, so we retry until
 	// we successfully populate the replicaset config.
-	var err error
-	for attempt := initiateAttemptStrategy.Start(); attempt.Next(); {
-		err = attemptInitiateMongoServer(p.DialInfo, p.MemberHostPort)
-		if err == nil {
-			logger.Infof("replica set initiated")
-			return err
-		}
-		if attempt.HasNext() {
-			logger.Debugf("replica set initiation failed, will retry: %v", err)
+	retryStrategy := initiateAttemptStrategy
```

Comment:
Since initiateAttemptStrategy isn't used else where, do we really need it?  Suggest initializing the retryStrategy like: https://github.com/juju/juju/blob/1eea923942403fe8d258d8a825d2da4c1d846744/caas/kubernetes/provider/bootstrap.go#L648-L660 is easier to read.

Reply:
This works great apart from the fact that NotifyFunc uses retryStrategy.Attempts

Which I'm just noticing now isn't actually set!